Zend/zend_vm_def.h

```diff
@@ -3751,17 +3751,20 @@ ZEND_VM_HOT_HANDLER(59, ZEND_INIT_FCALL_BY_NAME, ANY, CONST, NUM|CACHE_SLOT)
 {
 	USE_OPLINE
 	zend_function *fbc;
-	zval *function_name, *func;
 	zend_execute_data *call;
 
+	SAVE_OPLINE();
 	fbc = CACHED_PTR(opline->result.num);
 	if (UNEXPECTED(fbc == NULL)) {
-		function_name = (zval*)RT_CONSTANT(opline, opline->op2);
-		func = zend_hash_find_known_hash(EG(function_table), Z_STR_P(function_name+1));
-		if (UNEXPECTED(func == NULL)) {
-			ZEND_VM_DISPATCH_TO_HELPER(zend_undefined_function_helper);
+		zval *function_name = (zval*)RT_CONSTANT(opline, opline->op2);
+		/* Fetch lowercase name stored in the next literal slot */
+		fbc = zend_lookup_function_ex(Z_STR_P(function_name), Z_STR_P(function_name+1), /* use_autoload */ true);
+		if (UNEXPECTED(fbc == NULL)) {
+			if (EXPECTED(!EG(exception))) {
+				ZEND_VM_DISPATCH_TO_HELPER(zend_undefined_function_helper);
+			}
+			HANDLE_EXCEPTION();
```

Comment:
I would propose to try autoloading only after simple hash lookup.
This should completely eliminate performance degradation for existing code.
e.g.

```c
		function_name = (zval*)RT_CONSTANT(opline, opline->op2);
		func = zend_hash_find_known_hash(EG(function_table), Z_STR_P(function_name+1));
		if (UNEXPECTED(func == NULL)) {
                    func = zend_autoload_function(Z_STR_P(function_name), Z_STR_P(function_name+1));
		    if (UNEXPECTED(func == NULL)) {
			if (EXPECTED(!EG(exception))) {
				ZEND_VM_DISPATCH_TO_HELPER(zend_undefined_function_helper);
```

Reply:
Shouldn't this already be the case because of how I implemented ``zend_lookup_function_ex()``?

As it will first check the HashTable, and then call the autoloader if it hasn't found an entry.

Reply:
Your implementation introduces overhead of call to ``zend_lookup_function_ex()``, few additional checks and use of  ``zend_hash_find()`` instead of ``zend_hash_find_known_hash()``.